version/version.go

```diff
@@ -19,6 +14,33 @@ var (
 // ABCIVersion is the version of the ABCI library
 const ABCIVersion = "0.14.0"
 
+// Protocol is used for implementation agnostic versioning.
+type Protocol int64
```

Comment:
Is this easy to update in the future? If so I'm in full support if uint16. If not, then perhaps uint32. (uint64 is definitely overkill though imo)

Reply:
Agree it should be unsigned. I went for the larger 64-bit to give us more space to signal things so that versions in the future may not be just an incrementing integer, but somehow take advantage of the bit-array, kind of like https://github.com/bitcoin/bips/blob/master/bip-0009.mediawiki

Reply:
👍 for `uint64`.